acapy_plugin_toolbox/basicmessage.py

```diff
@@ -299,7 +301,19 @@ async def handle(self, context: RequestContext, responder: BaseResponder):
             localization=LocalizationDecorator(locale="en"),
         )
 
-        await responder.send(msg, connection_id=context.message.connection_id)
+        # Need to use a connection target, reply_to_verkey, and reply_from_verkey
+        # if we want to send to a socket
+        conn_mgr = ConnectionManager(session)
+        targets = await conn_mgr.get_connection_targets(connection=connection)
+        assert isinstance(targets, list)
+        assert targets
+        assert isinstance(targets[0], ConnectionTarget)
+        await responder.send(
+            msg,
+            connection_id=context.message.connection_id,
+            reply_to_verkey=targets[0].recipient_keys[0],
+            reply_from_verkey=targets[0].sender_key,
+        )
```

Comment:
I poked at the failures a bit more, and it seems that for whatever reason, the integration tests are failing during the response phase here. If I replace this with the old `responder.send` code, the tests appear to pass.

Reply:
After new changes had been made, this no longer seems to be the case and things are passing. Woohoo!